index.mjs

```diff
@@ -90,17 +88,23 @@ const sign_and_request = async({ sign, host, url, body }) => {
 		body: body_string,
 	})
 
-	const [ failed_response, successful_response ] = await catchify(post(url, {
-		headers: {
-			...headers,
-			authorization,
-		},
-		body: body_string,
-	}))
+	try {
+		const successful_response = await post(url, {
+			headers: {
+				...headers,
+				authorization,
+			},
+			body: body_string,
+		})
 
-	if (successful_response) {
-		return parse_xml_or_throw(await get_response_text(successful_response))
-	} else {
+		const response_string = await get_response_text(successful_response)
+
+		if (!response_string) {
+			throw new Error(`No body on response.  Status code was ${successful_response.statusCode} – response was "${response_string}"`)
+		}
+
+		return parse_xml_or_throw(response_string)
+	} catch (failed_response) {
```

Comment:
I'm all for this being a `try`/`catch`, but now there are three or four distinct things that could cause the `catch` on line 107 to be called:
- line 92 network or similar error could throw
- line 100 is probably fine but I'm not sure if `.text()` ever throws.
- line 103 throws
- line 106 can throw

Reply:
If the `catch` on line 111 is called, we'll likely lose the useful error message to line 113 or line 124

Reply:
I made PR #7 which hopefully adds the useful logging from this PR and reduces the number of code paths.